localstack/aws/forwarder.py

```diff
@@ -65,6 +67,36 @@ def HttpFallbackDispatcher(provider: object, forward_url_getter: Callable[[], st
     return ForwardingFallbackDispatcher(provider, get_request_forwarder_http(forward_url_getter))
 
 
+class ContextModifier(Protocol):
+    """
+    This protocol is used to apply a specific modification to a request context.
+    It does not modify the original context; it creates a newly one instead.
+    """
+
+    def modify_context(self, context: RequestContext) -> RequestContext:
+        ...
+
+
+class RegionContextModifier(ContextModifier):
+    """It modifies a `RequestContext` by changing its region."""
+
+    def __init__(self, region: str):
+        self.region = region
+
+    def modify_context(self, context: RequestContext) -> RequestContext:
+        _context = copy.copy(context)
```

Comment:
copying the request context like this is not safe. the context contains the original http request object, which has all sorts of IO internals that may break in unexpected ways when copying.